src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ParseJson.txt

```diff
@@ -87,6 +87,34 @@ Blank()
 >> Sum(Table(ParseJSON("[1, 2, 3, 4, 5]")), Value(Value))
 15
 
+>> ForAll(ParseJSON("[5, 6, 7]"), Value(ThisItem))
+[5,6,7]
+
+>> ForAll(ParseJSON("null"), Value(ThisItem))
+Blank()
+
+// We currently do not support null in any of the constructor functions, so this is an error
+>> ForAll(ParseJSON("[null, null, null]"), Abs(Value(ThisItem)))
+[Microsoft.PowerFx.Types.ErrorValue,Microsoft.PowerFx.Types.ErrorValue,Microsoft.PowerFx.Types.ErrorValue]
+
+>> ForAll(ParseJSON("[1]"), Value(ThisItem) / 0)
+[Microsoft.PowerFx.Types.ErrorValue]
+
+>> Sum(ForAll(ParseJSON("[5, 6, 7]"), Value(ThisItem)), Value)
+18
+
+>> Sum(ForAll(ParseJSON("[5, 6, 7]"), ThisItem), Value(Value))
+18
+
+>> ForAll(ParseJSON("[1,{""a"":false},""hello""]"), IfError(Text(ThisItem), IfError(Text(Value(ThisItem)), Text(Boolean(ThisItem.a)))))
+["1","false","hello"]
+
+>> ForAll(ParseJSON("[1,true,3,false]"), Value(ThisItem))
+[1,Microsoft.PowerFx.Types.ErrorValue,3,Microsoft.PowerFx.Types.ErrorValue]
+
+>> ForAll(ParseJSON("[1,2,3]"), If(Value(ThisItem) < 2, Value(ThisItem), ThisItem))
```

Comment:
>ForAll(ParseJSON("[1,2,3]"), If(Value(ThisItem) < 2, Value(ThisItem), ThisItem)) [](http://example.com/codeflow?start=3&length=80)

Another interesting case: `ForAll(ParseJSON("[1,2,3]"), ThisItem)` - is it supported? If so, what does it return?

Reply:
There is actually a test for this on line 123, the result is being passed into the Sum function. To answer your question, it returns the same thing as Table(ParseJSON(...))